### DISCUSSION TITLE
Themes store like themeforest

### DISCUSSION THREAD

**Discussion by hermesalvesbr** [apostrophecms/apostrophe#3653] [Q&A]
Would it be possible to have a store  themes for apostrophe? like wordpress themes.

Like themforest, but with themes for apostrophe?

**Comment by boutell** [comment]
Well, let's talk about how that might work. Most apostrophe developers today want to do their own frontend markup and styles, but it doesn't have to be that way. Community contributions could make this happen faster. 

If you wanted to ship an apostrophe theme today, you could do it as an apostrophe bundle in npm with “improve” modules for everything you wanted to override a template, setting or style of. So from a tech standpoint, it’s pretty close to possible now.

But a practical problem is that apostrophe doesn’t yet have default templates for, say, a home page and a default page, therefore our boilerplate projects do ship them, therefore there is always a “project level” that winds up beating the theme templates in most projects.

That is fixable of course by shipping default templates for pages and not just widgets, i.e. having a default theme in the core, and dropping those templates from the boilerplate.

Following this idea, other modules like events and blog would come with templates relying upon and extending the same css classes found in those templates.

Turning on a theme shipped this way would be easy enough, because if you add an npm apostrophe module that happens to be a bundle to your app.js, any improve modules within it automatically switch on.

As for a theme browser, we’ve been talking about an “executive” for single-site apostrophe that could handle reconfiguring the "apos object" on the fly without dropping connections if certain global doc properties change, i.e. the ability for an admin to edit settings we now think of as requiring a hardcoded option in the source code. That would also provide a way to toggle between theme bundles via the UI.

One issue is that styles and JavaScript that do ship with Apostrophe core are hard to shut off right now because of the way `ui/src` folders works It’s additive: index.scss and index.js can exist in both the base and an improvement. You currently can’t say “disable that particular core `index.scss` file completely and use this instead" currently. So a new feature might be needed there to specify such overrides.

But for the most part it's surprising how close we are to being ready to do this.

It would be helpful to know if community developers are interested in contributing to theme development.

**Reply by hermesalvesbr** [reply]
today wordpress is the leader in the market, and the ease of buying a theme for anything is a strong point. ghostcms is good but limited for the programmer.

Apostrophe needs to take off with themes to overcome Ghost and get closer to Wordpress.

if Apostrophe itself provides one or two complete themes and puts it in their extension store, it will already be a motivator for the community.

Not to mention that Apostrophe could sell themes developed by the community on its website.
